### PR TITLE
tests: counter: Enable nRF52810 testing

### PR DESCRIPTION
**Pull request by carlescufi**
Enable the counter capability in the board and add a .conf file for the
nRF52810 running on the PCA10040 board.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

File: tests/drivers/counter/counter_basic_api/boards/nrf52810_pca10040.conf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CONFIG_COUNTER_TIMER0=y
+CONFIG_COUNTER_TIMER0_PRESCALER=4
+CONFIG_COUNTER_TIMER1=y
+CONFIG_COUNTER_TIMER1_PRESCALER=4
+CONFIG_COUNTER_TIMER2=y
+CONFIG_COUNTER_TIMER2_PRESCALER=4
+CONFIG_COUNTER_RTC0=y
```
Comment (carlescufi) [inline tests/drivers/counter/counter_basic_api/boards/nrf52810_pca10040.conf]: The 52810 only has 2 RTCs

**Review by ioannisg — COMMENTED** [review]
Why do we need to test this one if we do not have the actual SOC in the dev kit?

File: tests/drivers/counter/counter_basic_api/boards/nrf52810_pca10040.conf (hunk 1 above)
Comment (jarz-nordic) [reply tests/drivers/counter/counter_basic_api/boards/nrf52810_pca10040.conf]: Is it intentional to not configure here RTC1 ?

Comment (nordic-krch) [reply tests/drivers/counter/counter_basic_api/boards/nrf52810_pca10040.conf]: RTC1 used by kernel

**Comment by anangl** [comment]
> Why do we need to test this one if we do not have the actual SOC in the dev kit?

To check if counter drivers in both flavors can be successfully built for the SoC, I guess.

**Review by anangl — APPROVED** [review]

**Review by jakub-uC — APPROVED** [review]

**Pull request merged**